src/vmm/src/persist.rs

```diff
@@ -657,6 +658,11 @@ fn send_uffd_handshake(
         uffd.as_raw_fd(),
     )?;
 
+    // We prevent Rust from closing the socket file descriptor to avoid a potential race condition
+    // between the mappings message and the connection shutdown. If the latter arrives at the UFFD
+    // handler first, the handler never sees the mappings.
+    forget(socket);
+
```

Comment:
Shouldn't we store this socket somewhere instead?

Reply:
what would be the advantage of that?

Reply:
I thought that in the future we were planning to reuse the socket.